websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,275 @@
+						url: location.href
+					}
+				];
+			} else if (document.title.startsWith("User:")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = "Viewing a user's page:";
+				presenceData.state = `@${
+					document.title.split("User:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View User",
+						url: location.href
+					}
+				];
+			} else if (
+				pathname.includes("/File:" || "/Sonic_Retro:General_disclaimer")
+			) {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (
+				(pathname === "/index.php" && document.title.includes("Search")) ||
+				pathname.includes("/Special:Search") ||
+				pathname.includes("/Special:Search#")
+			) {
+				presenceData.largeImageKey = "search";
+				if (document.title.startsWith("Search -"))
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for ${
+						document.title
+							.split("Search results for ")[1]
+							.split(" - Sonic Retro")[0]
+					}`;
+				}
+			} else if (pathname === "/index.php") {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (pathname.startsWith("User_talk:")) {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${document.title.split(" - Sonic Retro")[0]}`;
+				presenceData.buttons = [
+					{
+						label: "Read Article",
+						url: location.href
+					}
+				];
+			}
+			break;
+		}
+		case "forums.sonicretro.org": {
+			presenceData.smallImageKey = "forums_512";
+			presenceData.smallImageText = "forums.sonicretro.org";
+			if (search) {
+				const forumTitle = document.title.split(
+					" | Sonic and Sega Retro Forums"
+				)[0];
+				if (search.startsWith("?watched/fourums")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched forums";
+				} else if (search.startsWith("?watched/threads")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched forums";
+				} else if (search.startsWith("?finde-new/")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = "Browsing watched forums";
+				} else if (search.startsWith("?forums/")) {
+					presenceData.largeImageKey = "search";
+					presenceData.details = `Browsing throguht ${forumTitle.toLowerCase()}`;
+					presenceData.state = `"${forumTitle}"`;
+					presenceData.buttons = [
+						{
+							label: "View Forum",
+							url: location.href
+						}
+					];
+				} else if (search.startsWith("?threads/")) {
+					presenceData.largeImageKey = "reading";
+					presenceData.details = "Reading a thread";
+					presenceData.state = `"${forumTitle}"`;
```

Comment:
why

Reply:
I used to get errors when I didn't put a string and just had a variable. This presence has been sitting for a min so I haven't gone back and changed these.